### PR TITLE
backend/mongo: use GridFS to stream large zips

### PR DESCRIPTION
**Pull request by marwan-at-work**
fixes https://github.com/gomods/athens/issues/357

### CODE REVIEW

File: pkg/storage/mongo/getter.go [hunk 1]
```diff
@@ -24,9 +24,17 @@ func (s *ModuleStore) Get(ctx context.Context, module, vsn string) (*storage.Ver
 		}
 		return nil, err
 	}
+
+	zipName := s.gridFileName(module, vsn)
+	fs := s.s.DB(s.d).GridFS("fs")
+	f, err := fs.Open(zipName)
```
Comment (michalpristas) [inline pkg/storage/mongo/getter.go]: dont you need to close the file? i see gridfs has exported method for that

Comment (michalpristas) [reply pkg/storage/mongo/getter.go]: thanks @marwan-at-work missed that.

Comment (michalpristas) [reply pkg/storage/mongo/getter.go]: i'm thinking whether or not to rename ZIp -> ZipReadCloser 
so it's obvious you have to close it

Comment (marwan-at-work) [reply pkg/storage/mongo/getter.go]: Zip *is* a ReadCloser. As long as your code editor shows you the type, then it should be clear. But if you think being more verbose is better, happy to change it :)

Comment (michalpristas) [reply pkg/storage/mongo/getter.go]: no i guess there's no need. it's not a common practice so lets avoid it.

Comment (marwan-at-work) [reply pkg/storage/mongo/getter.go]: @michalpristas sounds good. Let me know if this PR is good on your side : )

**Pull request merged**